content/doc/book/pipeline/pipeline-best-practices.adoc

```diff
@@ -0,0 +1,95 @@
+---
+layout: section
+title: Pipeline Best Practices
+---
+ifdef::backend-html5[]
+:notitle:
+:description:
+:author: Alex Taylor
+:email: jenkinsci-docs@googlegroups.com
+:sectanchors:
+ifdef::env-github[:imagesdir: ../resources]
+ifndef::env-github[:imagesdir: ../../resources]
+:hide-uri-scheme:
+:toc:
+endif::[]
+
+= Applying best practices to Pipelines
+
```

Comment:
@alexanderrtaylor  
Here is where you should set expectations for what will be in this document, the structure of it, etc. 

For example: 
```suggestion

This guide provides a selection of best practices for pipelines. 
It is intended to point pipeline authors and maintainers is towards patterns that result in better Pipelines and away from pitfalls they might otherwise not be aware of.   

This guide is not meant to be an exhaustive list of all Pipeline best practices. While the guide provides a number of specific examples, is not meant to be a detailed "how-to".

This guide is arranged by area, guideline, then specific examples. 
```

And so on.  This will help people (including reviewers and future contributors) know what to expect.